src/apps/velodrome/optimism/velodrome.bribe.contract-position-fetcher.ts

```diff
@@ -0,0 +1,101 @@
+    ) {
+        super(appToolkit);
+    }
+    veTokenAddress = '0x9c7305eb78a432ced5c4d14cac27e8ed569a2e26';
+    veVoteAddress = '0x09236cff45047dbee6b921e00704bed6d6b8cf7e';
+    groupLabel = 'Bribe';
+
+    getContract(address: string): VelodromeBribe {
+        return this.contractFactory.velodromeBribe({ address, network: this.network });
+    }
+
+    async getDefinitions(): Promise<DefaultContractPositionDefinition[]> {
+        const { data } = await axios.get<{ data: VelodromeApiPairData[] }>('https://api.velodrome.finance/api/v1/pairs');
+        var definitions = new Array;
+        data.data.filter(v => !!v).filter(v => !!v.gauge).forEach(pool => {
+            const wBribeAddress = pool.gauge.wrapped_bribe_address;
+            if (wBribeAddress != null) {
+                definitions.push({ address: wBribeAddress });
+            }
+        }
+        );
+        return definitions;
+    }
+
+    async getTokenDefinitions({ contract }) {
+        const numRewards = Number(await contract.rewardsListLength());
+        const bribeTokens = await Promise.all(range(numRewards).map(async n => await contract.rewards(n)));
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(this.network);
+        var tokenDefinitions = new Array();
+        for (let token of bribeTokens) {
+            const tokenFound = baseTokens.find(p => p.address === token);
+            if (tokenFound) {
+                tokenDefinitions.push({
+                    metaType: MetaType.CLAIMABLE,
+                    address: token,
+                    network: this.network,
+                });
+            }
+        }
+        return tokenDefinitions;
+    }
+
+    async getLabel({ contractPosition }: GetDisplayPropsParams<VelodromeBribe>): Promise<string> {
+        return `${contractPosition.address} Bribe`;
+    }
+
+    async getTokenBalancesPerPosition({ address, contract, contractPosition }: GetTokenBalancesParams<VelodromeBribe>): Promise<BigNumberish[]> {
+        const multicall = this.appToolkit.getMulticall(this.network);
+        const escrow = multicall.wrap(this.contractFactory.velodromeVe({ address: this.veTokenAddress, network: this.network }));
+        const veCount = Number(await escrow.balanceOf(address));
+        if (veCount === 0) {
+            return [];
+        };
+        const veTokenIds = await Promise.all(range(veCount).map(async i => {
+            const tokenId = await escrow.tokenOfOwnerByIndex(address, i);
+            return tokenId;
+        }))
+        const bribeTokens = contractPosition.tokens;
+        var tokenBalances = new Array;
+        for (let bribeToken of bribeTokens) {
+            var tokenBalance = 0;
+            for (let veTokenId of veTokenIds) {
+                const balance = await contract.earned(bribeToken.address, veTokenId);
+                tokenBalance += Number(balance);
+            }
+            tokenBalances.push(tokenBalance);
+        }
+        return tokenBalances;
+    }
```

Comment:
I suggest you have a look at `EthereumChickenBondBondContractPositionFetcher` and how to deal with getBalances and getTokenBalancesPerPositions

Reply:
Are you suggesting that I override getBalances instead of getTokenBalancesPerPositions?
That would allow to call escrow.balanceOf(address) and escrow.tokenOfOwnerByIndex(address, i) only once per address (instead of numberOfPool times)

Reply:
Yes 💯

Reply:
Done